docs/developer/first_delivery.md

```diff
@@ -0,0 +1,241 @@
+Now that you've familiarized yourself with what Atomist can do, let's build our first delivery flow.
+
+In this flow we will listen to changes made to your README file. On a push to a repository, Atomist will detect if your README has changed, and print a message. In [Local Mode](/developer/local/), this will print to the terminal feed; in [Team Mode](/developer/team/), we will send the message to a chat channel.
+
+## Prerequisites
+
+Before starting, make sure that you have installed [the atomist CLI](/quick-start/) and created [a blank SDM](/developer/sdm/#creating-an-sdm-project). As well, since we'll be working in TypeScript, it's a good idea to [configure your editor](/developer/typescript/) to work with that toolchain. (Remember to run `npm install` in your SDM's folder.)
+
+You also need a local repository that you don't mind making dummy commits and pushes to. We'll be working with [atomist-seeds/express-es6-rest-api](https://github.com/atomist-seeds/express-es6-rest-api). Fork this repository online; to clone it locally and have Atomist's git hooks already configured, run the following command:
+
+```
+atomist clone https://github.com/<YOUR_GITHUB_HANDLE>/express-es6-rest-api
+```
+
+This clones the project to the folder defined by the `ATOMIST_ROOT` environment variable (which defaults to `$HOME/atomist/projects`).
+
+## Defining your first goal
+
+A [goal](https://docs.atomist.com/developer/goal/) is a function that we want to run during certain events or within specific projects. In this case, our goal is to print a message.
+
+Open an editor that points to your local SDM's folder. All of the work in this tutorial will take place in the _lib/machine_ directory. There should be just one file in there: _machine.ts_. This defines the behavior of your SDM: identifying which goals to run, which events to react to, and so on.
+
+For now, we are going to focus on having a new message print. Create a new file in this directory called _message.ts_. We'll start by importing the `goal` function from the `@atomist/sdm` package:
+
+``` typescript
+import { goal } from "@atomist/sdm";
+```
+
+The `goal` function takes two arguments: a [GoalDetails object](https://docs.atomist.com/developer/goaldetails/), which provides some metadata about our goal, and the executing function itself.
+
+Let's start by defining a skeleton of what the `goal` function needs:
+
+```typescript
+export const messageGoal = goal(
+  { displayName: "display name",
+    uniqueName: "unique-name",
+  },
+  async goalInvocation => {
+      // do what is needed
+      return { code: 0 };
```

Comment:
`{ code: 0 }` isn't needed here. That's the default if nothing is getting returned.